### ISSUE TITLE
Having trouble importing filter structure correctly

### ISSUE THREAD

**Issue by dom11990** [easyw/kicadStepUpMod#88]
Hey, same guy from here https://forum.kicad.info/t/new-way-to-import-dxf-outlines-to-footprint/28516/26
Unfortunately I didn't see your reply. I will give it another go with the divider structure. 

I'm having some trouble with another component now. I started with a step file because for some reason the dxf wouldn't import at all with the following message in the log: "ImpEx Dxf - ignore degenerate arc of circle". So, I used the step file and via projections arrived at a sketch of the filter which looks like so:
![image](https://user-images.githubusercontent.com/4545056/128628031-99bcb1ca-8e54-4fc3-bc62-fe57ced5ba98.png)

It would be dreamy if I could just get this down to two ports, the fat one on the left and the fat one on the right. When I export this structure as is, it seems to ignore the mitered corners:
![image](https://user-images.githubusercontent.com/4545056/128628066-0a893ada-349b-42d0-9c37-b2a2d16b1571.png)

Any idea what is causing this? You mentioned that blocks should automatically coalesce now correct? So I would not need to go through and delete the duplicate lines etc.

Step file is attached.
[hairpin2.zip](https://github.com/easyw/kicadStepUpMod/files/6950508/hairpin2.zip)

**Comment by dom11990** [comment]
If I change the pad to "Pads_Poly" the export fails and I get:

```
['REF**']
12:03:01  ['Value ']
12:03:12  exporting new footprint
12:03:12  saving to /home/dom/Documents/kt_kicad/Footprints/KT_Filter.pretty/Filter_Hairpin_19250MHz.kicad_mod
12:03:12  Group
12:03:12  Sketch
12:03:12  Text001
12:03:12  Text
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  2d closed path
12:03:12  normalized Poly
12:03:12  poly closed
12:03:12  missing reference pad for polyline pad
12:03:12  Running the Python command 'ksuToolsFootprintGen' failed:
Traceback (most recent call last):
  File "/home/dom/.FreeCAD/Mod/kicadStepUpMod/kicadStepUpCMD.py", line 1986, in Activated
    kicadStepUptools.PushFootprint()
  File "/home/dom/.FreeCAD/Mod/kicadStepUpMod/kicadStepUptools.py", line 16621, in PushFootprint
    export_footprint(name)
  File "/home/dom/.FreeCAD/Mod/kicadStepUpMod/kicadStepUptools.py", line 17641, in export_footprint
    npad=npad+os.linesep+createFpPad(mpad,offset,u'Poly', polypad_pos)
  File "/home/dom/.FreeCAD/Mod/kicadStepUpMod/kicadStepUptools.py", line 18731, in createFpPad
    stop

name 'stop' is not defined
```

**Comment by dom11990** [comment]
I believe I've solved it. When using the Pads_Poly it is necessary to include a circular element of arbitrary size so kicad knows where the trace should connect to the pad. Once I put circles in all the freestanding elements the error went away and I was able to create a compatible part. In case someone else stumbles on this, here my solution:
![image](https://user-images.githubusercontent.com/4545056/128814918-865224eb-e8e1-4e2f-b1ce-d83ded010bb5.png)
